ext/ftp/ftp.c

```diff
@@ -1917,11 +1915,12 @@ static void ftp_ssl_shutdown(ftpbuf_t *ftp, php_socket_t fd, SSL *ssl_handle) {
 /* }}} */
 
 /* {{{ data_close */
-databuf_t*
-data_close(ftpbuf_t *ftp, databuf_t *data)
+void
+data_close(ftpbuf_t *ftp)
 {
+	databuf_t *data = ftp->data;
```

Comment:
Maybe just add an assertion that ftp is not NULL (I know that it currently is always valid but an assertion never hurts)